src/generators/legacy-json/index.mjs

```diff
@@ -0,0 +1,68 @@
+'use strict';
+
+import { writeFile } from 'node:fs/promises';
+import { join } from 'node:path';
+import { groupNodesByModule } from '../../utils/generators.mjs';
+import buildSection from './utils/buildSection.mjs';
+
+/**
+ * This generator is responsible for generating the legacy JSON files for the
+ *  legacy API docs for retro-compatibility. It is to be replaced while we work
+ *  on the new schema for this file.
+ *
+ * This is a top-level generator, intaking the raw AST tree of the api docs.
+ * It generates JSON files to the specified output directory given by the
+ *  config.
+ *
+ * @typedef {Array<ApiDocMetadataEntry>} Input
+ *
+ * @type {import('../types.d.ts').GeneratorMetadata<Input, import('./types.d.ts').Section[]>}
+ */
+export default {
+  name: 'legacy-json',
+
+  version: '1.0.0',
+
+  description: 'Generates the legacy version of the JSON API docs.',
+
+  dependsOn: 'ast',
+
+  async generate(input, { output }) {
+    // This array holds all the generated values for each module
+    const generatedValues = [];
+
+    const groupedModules = groupNodesByModule(input);
+
+    // Gets the first nodes of each module, which is considered the "head"
+    const headNodes = input.filter(node => node.heading.depth === 1);
+
+    /**
+     * @param {ApiDocMetadataEntry} head
+     * @returns {import('./types.d.ts').ModuleSection}
+     */
+    const processModuleNodes = head => {
+      const nodes = groupedModules.get(head.api);
+
+      const section = buildSection(head, nodes);
+      generatedValues.push(section);
+
+      return section;
+    };
+
+    await Promise.all(
```

Comment:
Can't we use a for here? To avoid using Promise.all?

Reply:
This is what was originally done, then after an initial review, we put a `Promise.all` instead of a series of awaits in a loop.

Reply:
Am I stuttering O.o -- I still believe await would be better here for readability

Reply:
in terms of code readability, it's the same for me. But from what I've been able to understand in this kind of case it's better for performance to use `Promise.all`.

Reply:
>  I've been able to understand in this kind of case it's better for performance to use Promise.all.

Why?

Reply:
> Why?

Unfortunately I don't remember much about it, but I'd seen a blog post about it.

Reply:
https://github.com/nodejs/api-docs-tooling/pull/92#discussion_r1785364417

Reply:
Ah yes, but that would only be the case if there's threading. It is correct to say that with the Promise approach at least the Promises will all be dispatched in parallel, but they will be executed one per time.

So in the end the feeling might be that it is faster, although it might not be. I do believe await syntax would better suit here, but no strong feelings.